src/deps_deploy/maven_settings.clj

```diff
@@ -0,0 +1,135 @@
+(ns deps-deploy.maven-settings
+  "Functions to work with maven settings.xml and settings-security.xml."
+  (:require [clojure.java.io :as io]
+            [clojure.string :as str])
+  (:import [org.apache.maven.settings.io.xpp3 SettingsXpp3Reader]
+           [org.apache.maven.settings
+            Server
+            Settings]
+           [org.sonatype.plexus.components.cipher DefaultPlexusCipher]
+           [org.sonatype.plexus.components.sec.dispatcher
+            DefaultSecDispatcher
+            SecUtil]
+           [org.sonatype.plexus.components.sec.dispatcher.model SettingsSecurity]))
+
+(def default-settings-path (str (System/getProperty "user.home") "/.m2/settings.xml"))
+(def default-settings-security-path (str (System/getProperty "user.home") "/.m2/settings-security.xml"))
+
+;; Adapted from https://github.com/jelmerk/maven-settings-decoder/blob/master/src/main/java/org/github/jelmerk/maven/settings/Decoder.java
+
+(defn decode-password
+  "Decodes a password (got from settings.xml)."
+  ^String [^String encoded-password ^String key]
+  (-> (DefaultPlexusCipher.)
+      (.decryptDecorated encoded-password key)))
+
+(defn read-settings-security
+  "Reads settings-security.xml file into an ^SettingsSecurity object.
+   Defaults to $HOME/.m2/settings-security.xml ."
+  (^SettingsSecurity []
+   (read-settings-security nil))
```

Comment:
Rather than call this with `nil` you can call it with `default-settins-security-path` and you'll get rid of the `or` in the  other arity.